File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionIndexOf.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionIndexOf extends TransformerFunction {
    public TransformerFunctionIndexOf() {
        super(FunctionDescription.of(
            Map.of(
            "substring", ArgumentType.of(ArgType.String).position(0),
            "from", ArgumentType.of(ArgType.Number).position(1)
            )));
    }
    @Override
    public Object apply(FunctionContext context) {
        var str = context.getString(null);
        var of = context.getString("substring");
        var from = context.getInteger("from");
        if (str == null || of == null) {
            return -1;
        }

        return from != null ? str.indexOf(of, from) : str.indexOf(of);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionIndexOf.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { isEqual } from "../JsonHelpers";
import JsonElementStreamer from "../JsonElementStreamer";

class TransformerFunctionIndexOf extends TransformerFunction {
  constructor() {
    super({
      argsSet: [
        { name: "substring", type: ArgType.String },
        { name: "from", type: ArgType.Number },
      ],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    const of = await context.getString("substring");
    const from = await context.getInteger("from");
    if (str == null || of == null) {
      return -1;
    }
    return from != null ? str.indexOf(of, from) : str.indexOf(of);
  }
}

export default TransformerFunctionIndexOf;
